src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -125,20 +128,39 @@ def __init__(
             sampling_period <= size
         ), "The sampling period should be equal to or lower than the window size."
 
+        sampling = sampling_period
+        self._resampler: Resampler | None = None
+        self._avg_sender: Sender[Sample] | None = None
+        self._resampler_task: asyncio.Task[None] | None = None
+        if resampling_period is not None:
+            assert (
+                resampling_period >= sampling_period
+            ), "Resampling must be equal to or higher than sampling, only down-sampling supported."
```

Comment:
Could you add this limitation to the class docstring as well?  And probably also mention that resampling is also possible.

Reply:
Also, why the limitation?

Reply:
I understood there is not use cases for the `MovingWindow` to support up-sampling at the moment, I was wondering about it and asked in slack last week.
Our common/typical scenario involves combining 2 data streams from different sources, e.g. meters, where one stream has a higher sampling period. So implementing down-sampling helps to preserve the precision of the data.

So should I just update the docs? Or add support for up-sampling and update the docs?

Reply:
That makes sense,  it is fine to say for now that the best they can get is the same rate as the resampling actor.  I'd say we discuss up-sampling if someone asks for it.

Reply:
I removed the limitation as in the current patch the user can pass the ResamplerConfig, but I'm fine to revert it if we don't want that.

Reply:
I think it would be a good idea to remove it for now, because we don't see that there would be any use-cases for this.  And look into it later when we come across an actual use-case for it.

Reply:
rigth, this was already removed.